<!-- Create the launch file for ROS -->
<launch>
	<arg name="speed" default="1.0"/>
	<arg name="file_name" default="cluster_coda"/>
	<node name="generatore" pkg="simulator" type="generator.py" output="screen">
		<param name="node_name" value="generatore"/>
		<param name="next_element" value="cluster1"/>
		<param name="gen_freq" value="15"/>
		<param name="speed" value="$(arg speed)"/>
		<param name="num_messages" value="100"/>
		<param name="pause_time" value="2"/>
		<param name="group_size" value="25"/>
		<param name="event_type" value="test3"/>
	</node>

	<node name="cluster1" pkg="simulator" type="cluster.py" output="screen">
		<param name="node_name" value="cluster1"/>
		<param name="next_element" value="coda"/>
		<param name="modality" value="unisci"/>
		<param name="cluster_events" value="20"/>
		<param name="next_cluster_topic" value="cluster2"/>
		<param name="speed" value="$(arg speed)"/>
	</node>

	<node name="cluster2" pkg="simulator" type="cluster.py" output="screen">
		<param name="node_name" value="cluster2"/>
		<param name="next_element" value="switch"/>
		<param name="modality" value="dividi"/>
		<param name="speed" value="$(arg speed)"/>
	</node>

	<node name="coda" pkg="simulator" type="coda.py" output="screen">
		<param name="node_name" value="coda"/>
		<param name="next_element" value="cluster2"/>
		<param name="num_servers" value="1"/>
		<param name="server_time" value="5"/>
		<param name="speed" value="$(arg speed)"/>
		<param name="uncertanity" value="False"/>
	</node>

	<node name="switch" pkg="simulator" type="switch2.py" output="screen">
		<param name="node_name" value="switch"/>
		<param name="modality" value="split"/>
		<param name="split_attribute" value="test3"/>
		<param name="topic1" value="coda2"/>
		<param name="topic2" value="coda3"/>
		<param name="attribute_num" value="1"/>
	</node>

	<node name="coda2" pkg="simulator" type="coda.py" output="screen">
		<param name="node_name" value="coda2"/>
		<param name="next_element" value="end"/>
		<param name="num_servers" value="1"/>
		<param name="server_time" value="2"/>
		<param name="speed" value="$(arg speed)"/>
		<param name="uncertanity" value="False"/>
	</node>

	<node name="coda3" pkg="simulator" type="coda.py" output="screen">
		<param name="node_name" value="coda3"/>
		<param name="next_element" value="end"/>
		<param name="num_servers" value="1"/>
		<param name="server_time" value="1"/>
		<param name="speed" value="$(arg speed)"/>
		<param name="uncertanity" value="False"/>
	</node>

	<node name="end" pkg="simulator" type="end_node.py" output="screen">
		<param name="node_name" value="end"/>
		<param name="modality" value="events"/>
		<param name="speed" value="$(arg speed)"/>
		<param name="file_name" value="$(arg file_name)"/>
	</node>

</launch>


